### PR TITLE
[docs] add docs for new Xcode 16.1 image

### PR DESCRIPTION
**Pull request by szdziedzic**
# Why

Add docs for new image

# How

Add docs for new image

# Test Plan

Preview

# Checklist

- [ ] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [ ] Conforms with the [Documentation Writing Style Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)
- [ ] This diff will work correctly for `npx expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

**Comment by szdziedzic** [comment]
* **#32503** <a href="https://app.graphite.dev/github/pr/expo/expo/32503?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @szdziedzic and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: docs/pages/build-reference/infrastructure.mdx [hunk 1]
```diff
@@ -277,7 +277,25 @@ iOS builder VMs run on Mac mini hosts in an isolated environment. Every build ge
...
+- pnpm 9.12.3
+- npm 9.8.1
+- fastlane 2.225.0
+- CocoaPods 1.16.0
+- Ruby 3.2
+- node-gyp 10.2.0
+
+</Collapsible>
+
+#### `macos-sonoma-14.6-xcode-16.0`
```
Comment (amandeepmittal) [inline docs/pages/build-reference/infrastructure.mdx:298]: Is there supposed to be an SDK version listed here?

Comment (szdziedzic) [reply docs/pages/build-reference/infrastructure.mdx:298]: Do you mean the SDK tag? No, 16.0 is not tagged as any `sdk-xyz` at the moment, but 15.4 is.

Comment (amandeepmittal) [reply docs/pages/build-reference/infrastructure.mdx:298]: Yes. Thanks for the clarification!

**Review by amandeepmittal — APPROVED** [review]

**Pull request merged**